sdk/nextjs/src/server/initialize.ts

```diff
@@ -49,15 +54,49 @@ export const initialize = cache(
             ...jsClientOptions,
         })
 
-        let config = null
-        try {
-            config = await getBucketedConfig(
+        const getUserConfig = () => {
+            return getBucketedConfig(
                 configData.config,
                 configData.lastModified,
                 user,
                 options,
                 userAgent,
             )
+        }
+
+        let config = null
+        try {
+            if (configData.config.project.settings.optIn?.enabled) {
+                try {
+                    const hasOptInRecords = await hasOptInEnabled(
+                        user.user_id,
+                        sdkKey,
```

Comment:
should be calling this with the clientSDK key, since has enabled does not accept server keys